src/traceability/azure-traceability-helper.ts

```diff
@@ -77,32 +83,53 @@ async function createDeploymentResource(aksResourceContext: AksResourceContext,
   });
 }
 
-export async function addTraceability(): Promise<void> {
+export async function addTraceability(deployedManifestFiles: string[]): Promise<void> {
   const aksResourceContext = getAksResourceContext();
-  const deploymentReport = createDeploymentReport(aksResourceContext);
-  try {
+  if (aksResourceContext !== null) {
+    const deploymentReport = getDeploymentReport(aksResourceContext, deployedManifestFiles);
     const deploymentPayload = getDeploymentPayload(deploymentReport);
-    console.log(`Trying to create the deployment resource with payload: \n${JSON.stringify(deploymentPayload)}`);
-    const deploymentResource = await createDeploymentResource(aksResourceContext, deploymentPayload);
-    console.log(`Deployment resource created successfully. Deployment resource object: \n${JSON.stringify(deploymentResource)}`);
-  } catch (error) {
-    console.log(`Some error occured: ${error}`);
+    try {
+      console.log(`Trying to create the deployment resource with payload: \n${JSON.stringify(deploymentPayload)}`);
+      const deploymentResource = await createDeploymentResource(aksResourceContext, deploymentPayload);
+      console.log(`Deployment resource created successfully. Deployment resource object: \n${JSON.stringify(deploymentResource)}`);
+    } catch (error) {
+      console.log(`Some error occured: ${error}`);
+      return Promise.reject(error);
```

Comment:
Let's not make the deployment fail because of traceability.
Change the message to `Some error occured while creating the deployment resource for traceability: '${error}'` ?

Reply:
Remove `return Promise.reject(error);`?

Reply:
Are we not failing the run in this case?

Reply:
In my opinion, we should not. We are injecting traceability without user's input. We should not start failing existing workflows.
Maybe, we should add a `warning`. Something to discuss with the PMs? (I believe PMs had this ask to introduce this in the existing action version, and not require user's to start using a newer version).